rust/nix-expr/src/value.rs

```diff
@@ -19,26 +19,31 @@ pub enum ValueType {
     Null,
     Path,
     String,
-    Thunk,
     Unknown,
 }
 
-impl ValueType {
-    pub(crate) fn from_raw(raw: raw::ValueType) -> ValueType {
+#[derive(Eq, PartialEq, Debug)]
+pub enum ValueTypeOrThunk {
+    ValueType(ValueType),
+    Thunk,
+}
```

Comment:
I think it'd be ok to use `Option<ValueType>` for this. That way we don't have a nice "thunk" name, but `Option` has nice methods and can be passed to generic functions.

Reply:
there is the option of implementing `Into<Option<ValueType>> for ValueTypeOrThunk` too.

Reply:
Right, but that all seems a bit heavy for what's a really niche use case.
All normal code must always force thunks anyway, so it seems a bit over-engineered to me.